packages/permissions-kernel-snap/snap.manifest.ts

```diff
@@ -30,4 +33,10 @@ const manifest: SnapManifest = {
   manifestVersion: '0.1',
 };
 
+if (gatorSnapId) {
+  manifest.initialConnections = {
+    [gatorSnapId]: {},
+  };
+}
```

Comment:
Isn't this connection required even in production?

Why isn't it already included? Was it removed when we removed the lifecycle hooks originally?

Should we validate the GATOR_PERMISSIONS_PROVIDER_SNAP_ID and throw if it's not set (unless I'm wrong about it being required).